subprojects/files/src/test/groovy/org/gradle/api/internal/file/collections/GeneratedSingletonFileTreeTest.java

```diff
@@ -116,26 +72,25 @@ public void execute(OutputStream outputStream) {
     @Test
     public void doesNotOverwriteFileWhenGeneratedContentRemainsTheSame() throws InterruptedException {
         Action<OutputStream> action = getAction();
-        tree.add("path/file.txt", action);
+        MinimalFileTree tree = tree("path/file.txt", action);
 
-        assertVisits(tree, toList("path/file.txt"), toList("path"));
```

Comment:
I think we still would be visiting the `path` directory if we add a non-trivial relative path.

Reply:
brought this behaviour back